src/suites/cts/validation/vertex_input.spec.ts

```diff
@@ -359,6 +411,42 @@ g.test('check out of bounds condition on total number of vertex attributes', asy
   }
 });
 
+g.test('check out of bounds on number of vertex attributes across vertex buffers', async t => {
+  function getVertexBuffer(shaderLocation: number): GPUVertexBufferDescriptor {
+    return {
+      stride: 0,
+      attributeSet: [
+        {
+          shaderLocation,
+          format: 'float',
+        },
+      ],
+    };
+  }
+
+  const vertexBuffers: GPUVertexBufferDescriptor[] = [];
+  for (let i = 0; i < MAX_VERTEX_ATTRIBUTES; i++) {
+    vertexBuffers.push(getVertexBuffer(i));
+  }
+
+  {
+    // Control case, setting max vertex buffer number
+    const vertexInput: GPUVertexInputDescriptor = { vertexBuffers };
+    const descriptor = t.getDescriptor(vertexInput, VERTEX_SHADER_CODE_WITH_NO_INPUT);
+    t.device.createRenderPipeline(descriptor);
+  }
+  {
+    // Test vertex attribute number exceed the limit
+    vertexBuffers.push(getVertexBuffer(MAX_VERTEX_ATTRIBUTES));
```

Comment:
Doesn't this also push the number of vertex _buffers_ to max+1? If the intent is to test the max attributes then the extra attribute needs to be on an existing buffer.

Reply:
You are absolutely right. I should have caught this.